webapp-django/crashstats/crashstats/management/commands/verifyprocessed.py

```diff
@@ -27,24 +28,42 @@
 RAW_CRASH_PREFIX_TEMPLATE = "v2/raw_crash/%s/%s/"
 PROCESSED_CRASH_TEMPLATE = "v1/processed_crash/%s"
 
+# Number of seconds until we decide a worker has stalled
+WORKER_TIMEOUT = 10 * 60
+
 
 metrics = markus.get_metrics("cron.verifyprocessed")
 
 
-def check_crashids(entropy, boto_conn, bucket_name, date):
+def check_crashids(entropy_chunk, date):
     """Checks crash ids for a given entropy and date."""
-    bucket = boto_conn.get_bucket(bucket_name)
-
-    raw_crash_key_prefix = RAW_CRASH_PREFIX_TEMPLATE % (entropy, date)
+    s3_context = get_s3_context()
+    bucket = s3_context.config.bucket_name
+    s3_client = s3_context.client
 
     missing = []
-    for key_instance in bucket.list(raw_crash_key_prefix):
-        raw_crash_key = key_instance.key
-        crash_id = raw_crash_key.split("/")[-1]
+    for entropy in entropy_chunk:
+        raw_crash_key_prefix = RAW_CRASH_PREFIX_TEMPLATE % (entropy, date)
+
+        paginator = s3_client.get_paginator("list_objects_v2")
+        page_iterator = paginator.paginate(Bucket=bucket, Prefix=raw_crash_key_prefix)
 
-        processed_crash_key = bucket.get_key(PROCESSED_CRASH_TEMPLATE % crash_id)
-        if processed_crash_key is None:
-            missing.append(crash_id)
+        for page in page_iterator:
+            for item in page.get("Contents", []):
+                raw_crash_key = item["Key"]
+                crash_id = raw_crash_key.split("/")[-1]
+
+                try:
+                    s3_client.head_object(
+                        Bucket=bucket, Key=PROCESSED_CRASH_TEMPLATE % crash_id
+                    )
+                except s3_client.exceptions.ClientError as exc:
+                    # If we got back a 404: Not Found, then the processed crash isn't
+                    # there. If we got something else back, re-raise it.
+                    if "404" in str(exc):
```

Comment:
🤢 Maybe the Tecken version would be slightly better:

```
if exc.response["Error"]["Code"] == "404"
```

Reply:
Mmm... yes. It probably would be. I'll fix it.